src/robot/reporting/jsbuildingcontext.py

```diff
@@ -73,7 +73,8 @@ def create_link_target(self, msg):
 
     def check_expansion(self, kw):
         if self._expand_matcher is not None:
-            self._expand_matcher.match(kw)
+            if kw.passed:
+                self._expand_matcher.match(kw)
```

Comment:
1. I don't think not expanding keywords with SKIP status is a good idea. Keywords only get that status if they explicitly skip a test. That's pretty rare, so shouldn't cause extra overhead, and it could also be useful to be able to expand them. Should thus use `not kw.not_run` instead of `kw.passed`.
2. I think `ExpandKeywordMatcher` itself should make the decision about expanding. This `if` would thus rather belong there.

Reply:
1.+2.: ok, i agree. 
I updated the pull request and tested: Expanding also skipped keywords is safe.